engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobTimeoutTrigger.java

```diff
@@ -56,7 +56,7 @@ public void onPaused() {
 
   @Override
   public void onResumed() {
-    if (timer == null) {
+    if (timer == null && writer != null) { // only when recovered
```

Comment:
@saig0 this stuff I would like to discuss. What we actually would like to have, how should it behave. Are we expecting that when it was paused/resumed on replay that the listeners are called? Does it make sense ? Or maybe we don't want to call them, then we don't need that check here and we need to make it explicit in the inteface.